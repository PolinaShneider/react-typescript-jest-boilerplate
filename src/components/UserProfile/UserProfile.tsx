import React from 'react';
import style from './index.scss';

interface UserProfileProps {
  userName: string;
  userBio: string;
  handleClick: () => void;
}

export const UserProfile = ({ userBio, userName, handleClick }: UserProfileProps) => {
  return (
    <div className={style.userProfile}>
      <h1>{userName}</h1>
      <p data-testid="user-text" onClick={handleClick}>
        {userBio}
      </p>
    </div>
  );
};
