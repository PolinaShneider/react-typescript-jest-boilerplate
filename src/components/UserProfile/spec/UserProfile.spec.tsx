import React from 'react';
import renderer from 'react-test-renderer';

import { UserProfile } from '@components/UserProfile/UserProfile';

describe('UserPage component', () => {
  test('should render component properly', () => {
    // given
    const dummyComponentProps = {
      userName: 'Vasua Pupkin',
      userBio: 'coder',
      handleClick: () => console.log(123),
    };

    // when
    const componentRenderer = renderer.create(<UserProfile {...dummyComponentProps} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
