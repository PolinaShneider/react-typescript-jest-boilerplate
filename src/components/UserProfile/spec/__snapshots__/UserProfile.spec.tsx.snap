// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage component should render component properly 1`] = `
<div
  className="userProfile"
>
  <h1>
    Vasua Pupkin
  </h1>
  <p
    data-testid="user-text"
    onClick={[Function]}
  >
    coder
  </p>
</div>
`;
